.github/workflows/publish.yml

```diff
@@ -30,11 +32,24 @@ jobs:
       - run: npm run build
       - run: echo foundation.rust-lang.org > _site/CNAME
 
-      - name: deploy
-        uses: peaceiris/actions-gh-pages@v3
+      - uses: actions/upload-pages-artifact@v1.0.9
         with:
-          publish_dir: _site 
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          user_name: 'github-actions[bot]'
-          user_email: 'github-actions[bot]@users.noreply.github.com'
-          commit_message: ${{ github.event.head_commit.message }}
+          path: ./_site
+
+  deploy:
+    if: ${{ github.ref == 'refs/heads/main' }}
+
+    needs: build
+
+    permissions:
+      pages: write
+      id-token: write
+
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
```

Comment:
I guess this value gets populated automatically during the build process?

Reply:
yep, it refers to the `id: deployment` step below and the `actions/deploy-pages` action has a `page_url` output